src/scheduler/limits.c

```diff
@@ -2965,11 +2971,15 @@ check_max_user_res_soft(resource_resv **rr_arr, resource_resv *rr,
 		} else if (max_genuser_res_soft < used) {
 			if (rescts != NULL)
 				rescts->soft_limit_preempt_bit = preempt_bit;
-			return (preempt_bit);
+			rc = preempt_bit;
+		} else {
+			/* usage is under generic user soft limit, reset the preempt bit */
+			if (rescts != NULL)
+				rescts->soft_limit_preempt_bit = 0;
```

Comment:
In case of a specific limit, wouldn't control fall into the block - [L2961 ](https://github.com/PBSPro/pbspro/pull/1444/files/0e83e409f0e2730568309770e3718616086eb5f4#diff-4a1b68c3622f68d77b25310418c0e602R2961-R2970). Since we already check for specific limit, I thought this was the right place to put it.